crates/gen/src/com_interface.rs

```diff
@@ -138,14 +215,81 @@ impl ComInterface {
             impl #name {
                 #(#methods)*
             }
+            #conversions
         }
     }
 
     pub fn dependencies(&self) -> Vec<winmd::TypeDef> {
         self.methods
             .iter()
             .map(|method| method.signature.dependencies())
+            .chain(self.bases.iter().map(|base| base.dependencies()))
             .flatten()
             .collect()
     }
 }
+
+fn gen_constraint(method: &Method) -> TokenStream {
+    let mut tokens = Vec::new();
+
+    for (position, param) in method.signature.params.iter().enumerate() {
+        if !param.is_input || param.is_array {
+            continue;
+        }
+
+        match &param.kind {
+            TypeKind::IUnknown | TypeKind::Interface(_) => {
```

Comment:
nit: this might be easier to read as an `if let`

Reply:
This is going to need more match patterns as I finish up the type system support.